Less verbose integration test output

Anviking opened the pull request:
# Issue Number

#2119 


# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] More quiet stdout per default when running integration tests
- [x] Log full wallet, node and test-setup logs to `$TESTS_LOGDIR`, if set.
- [x] Upload `$TESTS_LOGDIR/*.log` as Buildkite artefacts

# Comments

- Regardless of `$TESTS_LOGDIR` setting, logs are still written to the temporary cluster directory as before.
- We can make hydra and GitHub Actions upload logs in a future PR. We might also be fine without it.

<img width="1458" alt="Skärmavbild 2020-09-10 kl  17 57 58" src="https://user-images.githubusercontent.com/304423/92758896-79597480-f38f-11ea-9394-501dc4608221.png">


<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

Anviking committed:
Remove wallet and node integration logs from stdout

And allow logging to a dedicated log directory by programatically
modifying the CM.Configurations.

Anviking committed:
Set and upload TESTS_LOGDIR from buildkite

Buildkite: Do --no-nix-pure setting through stack.yaml

Co-authored-by: Rodney Lorrimar <rodney.lorrimar@iohk.io>

Anviking committed:
Remove cartouches and make ClusterLog more neat

It now explains NO_POOLS=1 to you, and tells you if it is set or not.

Also disables two "expected warnings".

rvl committed:
nix: set TESTS_LOGDIR for integration tests on hydra

Anviking committed:
Lower MsgPoolGarbageCollectionEvent severity to Info

KtorZ commented on lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:
```diff
@@ -777,19 +796,37 @@ genConfig
+    let scribes = catMaybes
+            [ Just $ fileScribe ("cardano-node.log", severity)
+            , fileScribe . first T.pack <$> mExtraLogFile
+            ]
```
What's the use for two log files here?

Anviking replied:
It seemed drastic/weird to remove the existing node logs from the temporary directory.

Without setting additional environment variables, we can inspect the dbs, and the logs, in that directory, while the tests are running.

Thought it is also a bit weird to duplicate the logs 🤷‍♂️

KtorZ commented on lib/shelley/test/integration/Main.hs:
```diff
@@ -242,26 +246,29 @@ specWithServer (tr, tracers) = aroundAll withContext . after tearDown
     tr' = contramap MsgCluster tr
     onByron _ = pure ()
     afterFork dir _ = do
+        traceWith tr MsgSettingUpFacuet
```
*Facuet

KtorZ commented on lib/shelley/test/integration/Main.hs:
```diff
@@ -348,13 +370,19 @@ bracketTracer' tr name = bracketTracer (contramap (MsgBracket name) tr)
 walletMinSeverityFromEnv :: IO Severity
 walletMinSeverityFromEnv =
-    minSeverityFromEnv Info "CARDANO_WALLET_TRACING_MIN_SEVERITY"
+    minSeverityFromEnv Critical "CARDANO_WALLET_TRACING_MIN_SEVERITY"
```
Is it really sound? Sure we will have less logs but we still needs those logs.

Anviking replied:
This is the severity used for `Stdout`.

KtorZ committed:
fix typo in log constructor 'Facuet' --> 'Faucet'

KtorZ committed:
adjust code for taking new faucets to replace the empty faucet back (instead of hanging forever)

  This causes whatever test runs out of faucet to fail with a proper error message, instead of hanging forever.

KtorZ committed:
add moar faucets, and also increase the number of UTxO per batch when funding faucets

  So that we don't need to re-generate faucets of faucets
